components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediatorFactory.java

```diff
@@ -0,0 +1,283 @@
+                    "Unable to create the cache mediator. Unexpected element as the cache mediator configuration");
+        }
+
+        CacheMediator cache = new CacheMediator();
+
+        OMAttribute collectorAttr = elem.getAttribute(ATT_COLLECTOR);
+        if (collectorAttr != null && collectorAttr.getAttributeValue() != null) {
+            if ("true".equals(collectorAttr.getAttributeValue())) {
+                cache.setCollector(true);
+            } else if ("false".equals(collectorAttr.getAttributeValue())) {
+                cache.setCollector(false);
+
+                OMAttribute timeoutAttr = elem.getAttribute(ATT_TIMEOUT);
+                if (timeoutAttr != null && timeoutAttr.getAttributeValue() != null) {
+                    cache.setTimeout(Long.parseLong(timeoutAttr.getAttributeValue().trim()));
+                } else {
+                    cache.setTimeout(CachingConstants.DEFAULT_TIMEOUT);
+                }
+
+                OMAttribute maxMessageSizeAttr = elem.getAttribute(ATT_MAX_MSG_SIZE);
+                if (maxMessageSizeAttr != null && maxMessageSizeAttr.getAttributeValue() != null) {
+                    cache.setMaxMessageSize(Integer.parseInt(maxMessageSizeAttr.getAttributeValue().trim()));
+                } else {
+                    cache.setMaxMessageSize(-1);
+                }
+
+                String className = null;
+                OMElement protocolElem = elem.getFirstChildWithName(PROTOCOL_Q);
+                Map<String, Object> props = new HashMap<>();
+                if (protocolElem != null) {
+                    OMAttribute typeAttr = protocolElem.getAttribute(ATT_TYPE);
+
+                    if (typeAttr != null &&
+                            typeAttr.getAttributeValue() != null) {
+                        OMElement hashGeneratorElem = protocolElem.getFirstChildWithName(HASH_GENERATOR_Q);
+                        if (hashGeneratorElem != null) {
+                            className = hashGeneratorElem.getText();
+                        }
+
+                        String protocolType = typeAttr.getAttributeValue().toUpperCase().trim();
+                        cache.setProtocolType(protocolType);
+                        if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(protocolType)) {
+                            OMElement methodElem = protocolElem.getFirstChildWithName(HTTP_METHODS_TO_CACHE_Q);
+                            if (methodElem != null) {
+                                String[] methods = methodElem.getText().split(",");
+                                if (!"".equals(methods[0])) {
+                                    for (int i = 0; i < methods.length; i++) {
+                                        methods[i] = methods[i].toUpperCase().trim();
+                                        if (!(PassThroughConstants.HTTP_POST.equals(methods[i]) ||
+                                                PassThroughConstants.HTTP_GET.equals(methods[i]) ||
```

Comment:
In if condition, it would be better to have the variable first and have the constant in the equals bracket.

Reply:
It is usual practice to have the constant first and the variable in the equals bracket because it is possible that the variable could be null and hence cause a `NullPointerException`. What do you think?

Reply:
In which case, shouldn't we check for null? or in this scenario, it can be null?

Reply:
`methods` variable is populated like this: `String[] methods = methodElem.getText().split(",");`. Hence in this case it won't be null